tap_zendesk/client.py

```diff
@@ -129,8 +121,135 @@ def parse_response(self, response: requests.Response) -> Iterable[dict]:
         Yields:
             Each record from the source.
         """
-        # TODO: Parse response body and return a set of records.
-        yield from extract_jsonpath(self.records_jsonpath, input=response.json())
+        if not response.content:
+            self.logger.warning("Received empty response for URL: %s", response.url)
+            return
+
+        try:
+            response_json = response.json()
+            self.logger.debug(f"Response JSON: {response_json}")
+        except json.JSONDecodeError:
+            self.logger.error("Unable to decode JSON response: %s", response.text)
+            return
```

Comment:
Same here, don't we need to raise an exception here?

Reply:
I removed the `try/except` block so an exception will be raised if `response.json()` fails.